app/scripts/components/app/index.js

```diff
@@ -2,25 +2,42 @@ import React from 'react';
 import { connect } from 'react-redux';
 import { target, environment } from '../../config';
 import { displayCase } from '../../utils/format';
+import { getApiVersion, validateApiVersion } from '../../actions';
 
 import Header from './header';
 
 var App = React.createClass({
   displayName: 'App',
 
+  getInitialState: function () {
+    return {
+      api_version: ''
+    };
+  },
+
   propTypes: {
     children: React.PropTypes.object,
     dispatch: React.PropTypes.func,
     location: React.PropTypes.object,
     api: React.PropTypes.object
   },
 
+  componentDidMount: () => {
+    const { dispatch } = this.props;
+    dispatch(getApiVersion);
+    dispatch(validateApiVersion);
+  },
+
   render: function () {
     return (
       <div className='app'>
         { target !== 'cumulus' ? (
           <div className='app__target--container'>
             <h4 className='app__target'>{displayCase(target)} ({displayCase(environment)})</h4>
+            <h5 className='app__api_version'>
+              {`API Version: ${this.state.api_version}`}
+              { this.state.version_compat_error ? 'INCOMPATIBLE CUMULUS API _ PLEASE CHECK' : '' }
```

Comment:
I'm not seeing where this state is actually getting set? I think what you may want to do instead is read the API version from the `props` after connecting the API version state to this component.